### PR TITLE
Smarti in plain Rocket.Chat

### PR DESCRIPTION
**Pull request by mrsimpson**
This PR enables Smarti for all room types

### CODE REVIEW

**Comment by mrsimpson** [comment]
acl should include livechats if the user is an agent

**Comment by ThomasRoehl** [comment]
Missing:
- all i18 labels for settings in assistify
- rocket-chat-external-messages cache is missing therefor initialization of rooms for smarti is failing

Smarti Configuration in RocketChat
- Overall the configuration is very instable
- Sometimes the settings are not saved after clicking the save button
```
SmartiWidget.js:1 Uncaught SyntaxError: Unexpected identifier
    at smartiLoader.js:21
    at MethodInvoker._callback (meteor.js?hash=c9a34f92f28ede5a42eac467d3fa1763b624436a:1168)
    at MethodInvoker._maybeInvokeCallback (MethodInvoker.js:52)
    at MethodInvoker.receiveResult (MethodInvoker.js:71)
    at Connection._livedata_result (livedata_connection.js:1514)
    at Connection.onMessage (livedata_connection.js:1673)
    at browser.js:181
    at Array.forEach (<anonymous>)
    at ClientStream.forEachCallback (common.js:30)
    at SockJS.socket.onmessage (browser.js:180)
```

File: packages/assistify-ai/models/AssistifySmarti.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class AssistifySmarti extends RocketChat.models._Base {
```
Comment (mrsimpson) [inline packages/assistify-ai/models/AssistifySmarti.js]: @ThomasRoehl we're not using the rocketchat-external-messages-model anymore, but an own one.

**Comment by mrsimpson** [comment]
we're not using rocketchat-external-messages model and collection anymore, but an own one (`assistify-smarti`)

**Comment by ThomasRoehl** [comment]
https://github.com/assistify/Rocket.Chat/issues/433

**Comment by ThomasRoehl** [comment]
Tests are not running through yet because of html element issues

**Comment by ThomasRoehl** [comment]
The tests are all fixed.
But since I can not delete rooms, it is very unstable the more often you run it

**Comment by ruKurz** [comment]
remove the delta-synch button from the Smarti admin ui

**Comment by ruKurz** [comment]
Smarti does not analyze direct conversations (chats).
Therefore no suggestions can be provided in direct chats.
So the knowledge base should be deactivated in direct chats.

**Comment by ruKurz** [comment]
I was not able to test if the analysis results are changing, after writing new messages to the channel. This test uses the asynchronous Smarti call-back. And requires therefore an locally running Smarti or an integrated test env.

**Comment by ruKurz** [comment]
The knowledge base is now displayed in public and private rooms. The indexing has worked fine. As well suggestions (expert talks as well as Solr search results) are displayed now (after limiting the filter expressions) for private and public rooms.

**Comment by ruKurz** [comment]
A list of test scenarios would be great. Currently it feels like a random test approach that heavily depends on the ideas coming to my mind. We should create a list of test cases ...

**Comment by ruKurz** [comment]
In a regular channel (private/public) there will be no suggestions as long as you deactivate the Smarti topic filter. This problem can be solved by setting the topic to the name of the channel by default. Then all messages from threads (children) will be suggested in the channel (parent).

**Comment by ruKurz** [comment]
Smarti only suggests conversations that a user is allowed to see: Permission/Subscriptions should be tested by unit tests.

**Comment by mrsimpson** [comment]
@ruKurz additional tests should reflect the lifecycle of messages and rooms. Those are actually tested by the automated tests, but they anyway complete the set:

- [ ] Write a message, write a second message with the same text in another room - verify the message from the first room is found as related.
- [ ] Edit the first message. Re-open the Smarti-Widget in the second room. Verify that the found message is edited as well
- [ ] Delete the first message. Re-open the Smarti-Widget in the second room. Verify that the first message is gone

**Pull request merged**